projects/stargazer/plugins/other/radius/radius.cpp

```diff
@@ -17,3 +23,48 @@ RADIUS::RADIUS()
 {
 }
 
+int RADIUS::Start()
+{
+    m_thread = std::jthread([this](auto token){ Run(std::move(token)); });
+    return 0;
+}
+
+int RADIUS::Stop()
+{
+    std::lock_guard lock(m_mutex);
+
+    if (!m_thread.joinable())
+        return 0;
+
+    m_thread.request_stop();
+
+
+    if (isRunning)
+        m_thread.detach();
+    else
+        m_thread.join();
+
+    return 0;
+}
+
+int RADIUS::Run(std::stop_token token)
+{
+    std::lock_guard lock(m_mutex);
+    isRunning = true;
+
+    while (!token.stop_requested())
```

Comment:
This will not work. `io_service.run()` will run indefinitely and you won't reach the second iteration of this cycle. Let's just skip checking the `token` fro now.